virttest/virsh.py

```diff
@@ -621,6 +624,93 @@ def kosher_args(remote_ip, uri):
         return True not in all_false
 
 
+class EventNotFoundError(Exception):
+    """
+    Error when certain event cannot be found.
+    """
+
+    def __init__(self, details=''):
+        self.details = details
+        Exception.__init__(self)
+
+    def __str__(self):
+        return str(self.details)
+
+
+class EventTracker(object):
+
+    @staticmethod
+    def start_get_event(vm_name):
+        """
+        Use a virsh session with subcommand 'event' to catch events
+
+        :param vm_name: name of the vm to be catched
+        :return: the virsh session with 'event'
+        """
+        virsh_session = aexpect.ShellSession(VIRSH_EXEC)
+        event_cmd = 'event %s --all --loop' % vm_name
+        logging.info('Sending "%s" to virsh shell', event_cmd)
+        virsh_session.sendline(event_cmd)
+
+        return virsh_session
+
+    @staticmethod
+    def finish_get_event(virsh_session):
```

Comment:
In my understanding, what you want to expose is wait_event decorator , but wait_event doesn't call this method at all. 
Could you elaborate more what  use case finish_get_event want to achieve, how it interact with other methods?

Reply:
@chunfuwen yes, finish_get_event() has nothing to do with wait_for_event(). Please refer to https://github.com/autotest/tp-libvirt/pull/3371

Reply:
Better has usage guideline in your top part of your EventTracker class

Reply:
@chunfuwen oh? What kind of usage guideline?

Reply:
If I understand your API correctly, somehow it looks like below:

```
Usage: 

options:
  @wait_for_events decorator : what kind of this can achieve
  @start_wait_event, finish_wait_event:
      start_wait_event()
           .....
           func1()
      finish_wait_event()
```

Reply:
Oh? But I didn't see usage guideline in https://github.com/avocado-framework/avocado-vt/pull/2442 
So, seems function description is quite enough, agreed?
And I thought this thread of comments is about function finish_get_event() , right? Oddly we're discussing usage... really odd.

Reply:
> Oh? But I didn't see usage guideline in #2442
> So, seems function description is quite enough, agreed?
> And I thought this thread of comments is about function finish_get_event() , right? Oddly we're discussing usage... really odd.

@chloerh 
this is just enhancement to your exposed API.
You export two APIs and one decorator.
Until now I still need guest how we can use those API to achieve something except decorator on change_media

Reply:
@chunfuwen I'd be happy to explain to you. Would you like to tell me which part/line of code confuses you? And would you make your question more specific? It will help me understand which part is confusing you. Thanks.

Reply:
Chloerh has already updated the usage in the function so I think is comments should be resolved. Thanks